### PR TITLE
refacto: [NPM] Handle 0.0.0.0/0 ipblock and de-duplicated elements in except field

### PR DESCRIPTION
**Pull request by JungukCho**
**Reason for Change**:

To handle special IPBlock (0.0.0.0/0) and de-duplicated elements in except in translation logic.

**Issue Fixed**:

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [X] adds unit tests

**Notes**:
Instead of using `nomatch`, a better data structure to indicate whether `cidr` or `except` is needed for generic data planes.
Will improve it in next PR with revising data structure.
In addition, 0.0.0.0/0 handling was not decided yet in windows dataplane since windows does not support `nomatch` for `cidr` yet. 

Related previous PR:
1. [NPM] General translation logic for linux and windows #1055
2. refactor: [NPM] parsing label selector for general translation logic #1077
3. refactor: [NPM] General translation logic (mainly clean-up codes and correct bugs) #1105
4. refactor: [NPM] General translation logic for egress and ingress #1106

### CODE REVIEW

File: npm/pkg/controlplane/translation/translatePolicy.go [hunk 1]
```diff
@@ -127,17 +122,63 @@ func ipBlockSetName(policyName, ns string, direction policies.Direction, ipBlock
...
+	// Ipset doesn't allow 0.0.0.0/0 to be added.
+	// A solution is split 0.0.0.0/0 in half which convert to 0.0.0.0/1 and 128.0.0.0/1.
+	// splitCIDRSet is used to handle case where IPBlock has "0.0.0.0/0" in CIDR and "0.0.0.0/1" or "128.0.0.0/1"  in Except.
+	// splitCIDRSet has two entries ("0.0.0.0/1" and "128.0.0.0/1") as key.
+	splitCIDRLen := 2
+	splitCIDRSet := make(map[string]int, splitCIDRLen)
+	if ipBlockRule.CIDR == "0.0.0.0/0" {
+		// two cidrs (0.0.0.0/1 and 128.0.0.0/1) for 0.0.0.0/0 + except.
+		members = make([]string, lenOfDeDupExcepts+splitCIDRLen)
+		// in case of 0.0.0.0/0, 0.0.0.0/1 or 0.0.0.0/1nomatch comes eariler than 128.0.0.0/1 or 128.0.0.0/1nomatch.
```
Comment (huntergregory) [inline npm/pkg/controlplane/translation/translatePolicy.go:158]: nitpick: missing a space or newline

File: npm/pkg/controlplane/translation/translatePolicy.go [hunk 2]
```diff
@@ -127,17 +122,63 @@ func ipBlockSetName(policyName, ns string, direction policies.Direction, ipBlock
...
+		members = make([]string, lenOfDeDupExcepts+1)
+		members[indexOfMembers] = ipBlockRule.CIDR
+		indexOfMembers++
+	}
+
+	for i := 0; i < lenOfDeDupExcepts; i++ {
+		except := deDupExcepts[i]
+		if splitCIDRIndex, exist := splitCIDRSet[except]; exist {
+			// replace stored splitCIDR with "nomatch" option
+			members[splitCIDRIndex] = except + util.IpsetNomatch
```
Comment (huntergregory) [inline npm/pkg/controlplane/translation/translatePolicy.go:176]: we will have to update dataplane to be able to understand a cidr block like "1.2.3.4/16nomatch"

File: npm/pkg/controlplane/translation/translatePolicy_test.go [hunk 3]
```diff
@@ -367,6 +367,59 @@ func TestIPBlockIPSet(t *testing.T) {
...
+				Except: []string{"0.0.0.0/1", "128.0.0.0/1"},
+			},
+			translatedIPSet: ipsets.NewTranslatedIPSet("test-in-ns-default-0IN", ipsets.CIDRBlocks, []string{"0.0.0.0/1nomatch", "128.0.0.0/1nomatch"}...),
+		},
+		{
+			name:        "0.0.0.0/0 cidr and duplicated 0.0.0.0/1 and 128.0.0.0/1 except",
+			ipBlockInfo: createIPBlockInfo("test", "default", policies.Ingress, policies.SrcMatch, 0),
+			ipBlockRule: &networkingv1.IPBlock{
+				CIDR:   "0.0.0.0/0",
+				Except: []string{"0.0.0.0/1", "128.0.0.0/1", "128.0.0.0/1"},
```
Comment (huntergregory) [inline npm/pkg/controlplane/translation/translatePolicy_test.go:419]: by the test name, did you mean to have two "0.0.0.0/1" in the excepts?

**Review by huntergregory — COMMENTED** [review]
lgtm

File: npm/pkg/controlplane/translation/translatePolicy_test.go (hunk 3 above)
Comment (JungukCho) [reply npm/pkg/controlplane/translation/translatePolicy_test.go:447]: Reorganize UTs for better understanding.

File: npm/pkg/controlplane/translation/translatePolicy.go (hunk 2 above)
Comment (JungukCho) [reply npm/pkg/controlplane/translation/translatePolicy.go:176]: Good catch. Took the ad-hoc approach for linux now, but we will refactor this in next PR if you are ok.

File: npm/pkg/controlplane/translation/translatePolicy.go (hunk 1 above)
Comment (JungukCho) [reply npm/pkg/controlplane/translation/translatePolicy.go]: The comment was about `0.0.0.0/0nomatch'. Changed it to `0.0.0.0/0 nomatch` for better management in comments and codes as well.

**Review by huntergregory — APPROVED** [review]
just noting that we'll revise data structures to incorporate nomatch approach in a followup PR. We will also need to do integration testing to make sure it works correctly for Linux `ipset` command

**Pull request merged**